chore: Replace use of flakeheaven with Ruff for pep8-naming

nkclemson opened the pull request:
Closes #8402

 - [ ] squash git commits

Commit:
Replace use of flakeheaven with Ruff for pep8-naming #8402

Commit:
pep8-naming: allow mixCase variable (N815) on class scope #8402

Commit:
ref upper case variable name to lower case #8402

edgarrmondragon commented:
@nkclemson Ping me when this is ready for a review. Also, don't feel the need to squash commits  before marking this as ready since we do merge squashes anyway :)

nkclemson commented:
Hi @edgarrmondragon, I think this is complete now. I have added the `"N815" - mixedCase variable in class scope (constants, method names)` exemption, as this is in accordance with [Guido's recommendation](https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations). Initially, I thought that the lint error I encountered in test_logging_utils.py was N815, but I later realized that those all-caps variable names are inside methods, so I fixed it. let me know if I need to update anything.

edgarrmondragon commented on pyproject.toml:
```diff
@@ -586,6 +579,8 @@ exclude = [
 ignore = [
   "ANN101", # Missing type annotation for `self` in method
   "ANN102", # Missing type annotation for `cls` in class method
+  "N815",   # mixedCase variable in class scope (constants, method names)
```
I don't think this is currently triggered anywhere in the codebase

```suggestion
```

nkclemson replied:
yep. This isn't triggered. I am removing this.

edgarrmondragon commented on pyproject.toml:
```diff
@@ -625,6 +621,9 @@ unfixable = [
 "tests/**" = [
   "S101", # Allow 'assert' in tests
 ]
+"src/meltano/migrations/versions/**" = [
+  "N806", # Allow uppercase variable names
+]
```
I think this is covered by the `exclude` setting?

```suggestion
```

nkclemson replied:
Yep.

Commit:
removed N*** code from ignore list for mixCase and UPPER_CASE

edgarrmondragon reviewed (approved):
Thanks a ton @nkclemson!
